drivers/mspi/mspi_dw.c

```diff
@@ -112,6 +113,11 @@ struct mspi_dw_config {
 	uint8_t max_queued_dummy_bytes;
 	uint8_t tx_fifo_threshold;
 	uint8_t rx_fifo_threshold;
+#ifdef CONFIG_MSPI_DMA
+	uint8_t dma_tx_data_level;
+	uint8_t dma_rx_data_level;
+#endif
+	void const *vendor_specific_data;
```

Comment:
This pointer should stay `void *` as the pointed object does not necessarily must be read-only in all cases. Instead, use `(void *)` casting in the `VENDOR_SPECIFIC_DATA_GET()` definition for `nordic_nrf_qspi_v2`.